pkg/actuators/client/client.go

```diff
@@ -20,86 +20,419 @@ import (
 	"fmt"
 
 	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
 	"github.com/IBM/vpc-go-sdk/vpcv1"
+	ibmcloudproviderv1 "github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider/v1beta1"
 )
 
 // Client is a wrapper object for IBM SDK clients
 type Client interface {
-	InstanceGet(instanceID string) (*vpcv1.Instance, error)
-	// InstanceListAll()
-	// InstanceCreate()
-	// InstanceDelete()
-	// InstanceUpdate()
-	CustomImageGetID(imageName string, regionName string) (*vpcv1.Image, error)
+	// Instances functions
+	InstanceGetByID(instanceID string) (*vpcv1.Instance, error)
+	InstanceExistsByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (bool, error)
+	InstanceGetByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (*vpcv1.Instance, error)
+	InstanceDeleteByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) error
+	InstanceCreate(machineName string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec, userData string) (*vpcv1.Instance, error)
+
+	// Helper functions
+	GetCustomImageByName(imageName string, resourceGroupID string) (string, error)
+	GetVPCIDByName(vpcName string, resourceGroupID string) (string, error)
+	GetResourceGroupIDByName(resourceGroupName string) (string, error)
+	GetSubnetIDbyName(subnetName string, resourceGroupID string) (string, error)
+	GetSecurityGroupsByName(securityGroupNames []string, resourceGroupID string, vpcID string) ([]vpcv1.SecurityGroupIdentityIntf, error)
 }
 
 // ibmCloudClient makes call to IBM Cloud APIs
 type ibmCloudClient struct {
-	vpcService *vpcv1.VpcV1
+	vpcService                *vpcv1.VpcV1
+	resourceManagerService    *resourcemanagerv2.ResourceManagerV2
+	resourceControllerService *resourcecontrollerv2.ResourceControllerV2
 }
 
 // IbmcloudClientBuilderFuncType is function type for building ibm cloud client
 type IbmcloudClientBuilderFuncType func(credentialVal string) (Client, error)
 
 // NewClient initilizes a new validated client
 func NewClient(credentialVal string) (Client, error) {
-	authenticator := &core.IamAuthenticator{
-		ApiKey: credentialVal,
-	}
 
-	options := &vpcv1.VpcV1Options{
-		Authenticator: authenticator,
+	// IC Virtual Private Cloud (VPC) API
+	vpcService, err := vpcv1.NewVpcV1(&vpcv1.VpcV1Options{
+		Authenticator: &core.IamAuthenticator{
```

Comment:
Might be worth consolidating IamAuthenticator since the same credentials are passed to more than one service.
```
authenticator := &core.IamAuthenticator{ ApiKey: credentialVal }
vpcService, err := vpcv1.NewVpcV1(&vpcv1.VpcV1Options{
	Authenticator: authenticator,
})
```

Reply:
will change it back to how it was :)